iam.tf

```diff
@@ -0,0 +1,882 @@
+}
+
+
+
+data "aws_iam_policy_document" "ch_write_logs" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      data.terraform_remote_state.security-tools.outputs.logstore_bucket.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject*",
+      "s3:PutObject*",
+
+    ]
+
+    resources = [
+      "${data.terraform_remote_state.security-tools.outputs.logstore_bucket.arn}/${local.s3_log_prefix}",
+    ]
+  }
+}
+
+resource "aws_iam_policy" "ch_write_logs" {
+  name        = "cHWriteLogs"
+  description = "Allow writing of CH logs"
+  policy      = data.aws_iam_policy_document.ch_write_logs.json
+}
+
+resource "aws_iam_role_policy_attachment" "ch_write_logs" {
+  role       = aws_iam_role.ch_role_for_instance_profile.name
+  policy_arn = aws_iam_policy.ch_write_logs.arn
+}
+
+data "aws_iam_policy_document" "ch_metadata_change" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:ModifyInstanceMetadataOptions",
+      "ec2:*Tags",
+    ]
+
+    resources = [
+      "arn:aws:ec2:${var.region}:${local.account[local.environment]}:instance/*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "cloudwatch:PutMetricData",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "ec2:DescribeInstanceStatus",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "ds:CreateComputer",
+      "ds:DescribeDirectories",
+    ]
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+    ]
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      "arn:aws:s3:::eu-west-2.elasticmapreduce",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:Get*",
+      "s3:List*",
+    ]
+
+    resources = [
+      "arn:aws:s3:::eu-west-2.elasticmapreduce/libs/script-runner/*",
+    ]
+  }
+}
+
+resource "aws_iam_policy" "ch_metadata_change" {
+  name        = "CHMetadataOptions"
+  description = "Allow editing of Metadata Options"
+  policy      = data.aws_iam_policy_document.ch_metadata_change.json
+}
+
+resource "aws_iam_role_policy_attachment" "ch_instance_profile_role_metadata_change" {
+  role       = aws_iam_role.ch_role_for_instance_profile.name
+  policy_arn = aws_iam_policy.ch_metadata_change.arn
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_for_ssm_attachment" {
+  role       = aws_iam_role.ch_role_for_instance_profile.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
```

Comment:
AmazonEC2RoleforSSM is deprecated & grants sweeping access to S3, we should change this to AmazonSSMManagedInstanceCore.  If any issues are apparent after replacing the role, they are likely due to S3 permissions that need to be added